client/telemetry/src/lib.rs

```diff
@@ -16,339 +16,486 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! Telemetry utilities.
+//! To start using this module, please initialize the global logger from `sc-tracing`. This will
```

Comment:
I would really like to have these docs more a high level description on how the system works.

This means:
- How does it work with the span
- How does the communication works with the telemetry worker
- etc
- I just imagine some sort of "diagram" explaining it :D 

However, I will not make this a hard requirement to merge this.

Reply:
Does this look better? acf1ec852

Reply:
More doc bc3613628